packages/app/src/app/pages/common/UserMenu/index.tsx

```diff
@@ -1,44 +1,51 @@
-import * as React from 'react';
 import Relative from '@codesandbox/common/lib/components/Relative';
-import { inject, hooksObserver } from 'app/componentConnectors';
+import React, { FunctionComponent } from 'react';
+
 import { HoverMenu } from 'app/components/HoverMenu';
-import { Menu } from './Menu';
+import { useOvermind } from 'app/overmind';
+
 import { ClickableContainer, ProfileImage } from './elements';
+import { Menu } from './Menu';
+
+export const UserMenu: FunctionComponent = () => {
+  const {
+    actions: {
+      userMenuClosed,
+      modalOpened,
+      signOutClicked,
+      userMenuOpened,
+      files: { gotUploadedFiles },
+    },
+    state: { user, userMenuOpen },
+  } = useOvermind();
+
+  return (
+    <Relative>
+      <ClickableContainer onClick={userMenuOpened}>
+        <ProfileImage
+          alt={user.username}
+          width={30}
+          height={30}
+          src={user.avatarUrl}
+        />
+      </ClickableContainer>
 
-export const UserMenu = inject('store', 'signals')(
-  hooksObserver(
-    ({
-      store: { user, userMenuOpen },
-      signals: {
-        userMenuClosed,
-        modalOpened,
-        signOutClicked,
-        userMenuOpened,
-        files,
-      },
-    }) => (
-      <Relative>
-        <ClickableContainer onClick={userMenuOpened}>
-          <ProfileImage
-            alt={user.username}
-            width={30}
-            height={30}
-            src={user.avatarUrl}
+      {userMenuOpen && (
+        <HoverMenu onClose={() => userMenuClosed()}>
+          <Menu
+            openPreferences={() =>
+              modalOpened({ message: null, modal: 'preferences' })
```

Comment:
@christianalfoni Just like https://github.com/codesandbox/codesandbox-client/pull/2544#discussion_r328837587

Maybe make `message` optional? 🤔

Reply:
Yeah, we should make that optional and also the "modal" should be typed explicitly with types of modals you can open, now I believe it is a string? Would have to search codebase to find all possible modals though. 

Can maybe do that in a new PR? :)

Reply:
We made it optional here: https://github.com/codesandbox/codesandbox-client/pull/2553 , but yeah, the explicit typing of modal types can still be a new PR :)

Reply:
I'll make a new PR for the `modal` names 😉

Reply:
PR created: https://github.com/codesandbox/codesandbox-client/pull/2554